src/Compilers/Core/Portable/CodeGen/SequencePointList.cs

```diff
@@ -178,13 +170,34 @@ private static OffsetAndSpan[] GetSubArray(ArrayBuilder<RawSequencePoint> seqPoi
 
                         if (lastDebugDocument != null)
                         {
+                            int startLine = (fileLinePositionSpan.StartLinePosition.Line == -1) ? 0 : fileLinePositionSpan.StartLinePosition.Line + 1;
+                            int endLine = (fileLinePositionSpan.EndLinePosition.Line == -1) ? 0 : fileLinePositionSpan.EndLinePosition.Line + 1;
+                            int startColumn = fileLinePositionSpan.StartLinePosition.Character + 1;
+                            int endColumn = fileLinePositionSpan.EndLinePosition.Character + 1;
+
+                            // Trim column number if necessary.
+                            // Column must be in range [0, 0xffff) and end column must be greater than start column if on the same line.
+                            // The Portable PDB specifies 0x10000, but System.Reflection.Metadata reader has an off-by-one error.
+                            // Windows PDBs allow the same range.
+                            const int MaxColumn = ushort.MaxValue - 1;
+
+                            if (startColumn > MaxColumn)
```

Comment:
@tmat Should this check be `>=` instead? Per the above comment endColumn must be greater than startColumn if on the same line. But in the case where startColumn == MaxColumn and endColumn > MaxColumn, looks like we would end up with statColumn == endColumn (== MaxColumn) after the 2nd if block below...